api/v1/helpers/verbs/doPatch.js

```diff
@@ -72,7 +72,9 @@ function doPatch(req, res, next, props) {
 
         u.patchJsonArrayFields(o, requestBody, props);
         u.patchArrayFields(o, requestBody, props);
-        return o.update(requestBody);
+        return (props.modelName === 'Generator') ?
+          o.updateWithCollectors(requestBody, u.whereClauseForNameInArr) :
+          o.update(requestBody);
```

Comment:
Do you want to move this to the controller?

Reply:
Will change it in the doPatch pull request. Let's leave it here for now. The PR is already getting large.